eventlet/wsgi.py

```diff
@@ -694,16 +697,33 @@ def get_environ(self):
         env['REMOTE_PORT'] = str(client_addr[1])
         env['GATEWAY_INTERFACE'] = 'CGI/1.1'
 
-        try:
+        if six.PY2:
             headers = self.headers.headers
-        except AttributeError:
-            headers = self.headers._headers
-        else:
             headers = [h.split(':', 1) for h in headers]
+        else:
+            headers = self.headers._headers
+            payload = self.headers.get_payload()
```

Comment:
Oh jeeze -- apparently this may return a `str` _or_ a `list` of one or more messages (if `Content-Type` is `message/rfc822`).

:disappointed: